### DISCUSSION TITLE
When it makes sense to use lava link instead of using the integrated PCM Audio Source?

### DISCUSSION THREAD

**Discussion by Luc1412** [Rapptz/discord.py#6988] [Q&A] [answered]
I'm currently using a customized PCMVolumeTransformer with FFmpegPCMAudio. I have following requirements:
- Play an high quality MP3 with an average length of 1h30min
- Start the file at a specific timestamp (with a fast loading time) + get the current progress
- Handle at least 100 players at the same time 

Does it makes sense to switch to lavalink? (with wavelink)

When it generally makes sense to switch? What are the advantages?

**Comment by trevorflahardy** [comment, marked as answer]
Heya!

Switching to a Lavalink server, specifically using [Wavelink](https://github.com/PythonistaGuild/Wavelink) can easily handle 100+ players listening to high quality MP3 audio. Using Wavelink's built in functions, you can easily get the specific timestamp and current progress in the current track, song, or playlist. If I'm not mistaken, either Rhythm or Groovy Bot use a Lavalink server (not Python).

Let's run through a quick implementation for fetching current progress and timestamp.

```python
@bot.command()
async def timestamp(ctx):
    player = bot.wavelink.get_player(ctx.guild.id)
    pos = player.position  # The players seek position in the currently playing track in milliseconds. 
    
    # In order to get total track time, you would need to keep track of the current Track for each player.
    # You can do this with a class or something simple:
    current_track = self.tracks[ctx.guild.id].length  # The duration of the track in milliseconds.
```
You can do some simple maths from there to handle what you need. If you take a look at the [advanced example](https://github.com/PythonistaGuild/Wavelink/blob/master/examples/playlist.py) from Wavelink, you can see they use a **MusicController** to handle everything (which would make your current track much easier to get). Using FFmpegPCMAudio with a customized PCMVolumeTransformer, although coded well, still has the tendency to break or have unreliable audio / playback. I would recommend testing out Wavelink and then going from there.
